e2e/container/kill_test.go

```diff
@@ -0,0 +1,52 @@
+package container
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/cli/e2e/internal/fixtures"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/gotestyourself/gotestyourself/poll"
+)
+
+func TestKillContainer(t *testing.T) {
+	containerID := runBackgroundTop(t)
+
+	// Kill with SIGTERM should kill the process
+	result := icmd.RunCmd(
+		icmd.Command("docker", "kill", "-s", "SIGTERM", containerID),
+	)
+
+	result.Assert(t, icmd.Success)
+	poll.WaitOn(t, containerStatus(t, containerID, "exited"), poll.WithDelay(1*time.Second), poll.WithTimeout(10*time.Second))
```

Comment:
Won't the container be in the exited state before `kill` returns?

If not, could we use a delay of 100ms ?

Reply:
Thinking about it again, it definitely wont have that state before `kill` returns, so maybe just a shorter delay (like 100ms) ?

Reply:
yep, `1s` might be too long for sure 😅